teps/0053-tekton-catalog-pipeline-organization.md

```diff
@@ -0,0 +1,477 @@
+
+### Versioning Resources
+
+Resources definition evolves across time. A new version of the image used is published, a new feature in tektoncd/pipeline is available, or a new feature / behaviour is added. On the other hand, the catalog should guarantee as much stability as possible for its user.
+
+A version is an identifiable information of a resource along with its kind (Task, Pipeline) and name, and it should be present in the definition of the resource. The location of a resource in the catalog should be computable given its Kind, Name and the Version. e.g. `/{kind}/{name}/{version}/{name}.yaml` - `pipeline/<pipeline-name>/0.5/<pipeline-name>.yaml`
+
+Adding a new version of a resource should be considered as a release of that resource. Here are some cases where a version bump for pipeline is necessary
+
+- Task used in the pipeline have new image
+
+  - For images that we own we should only bump the version if the new version breaks CI or if release notes include changes on the Tekton image
+  - For images that we do not own: it should be the responsibility of the author of the task to verify if changing an image version requires a new Task version
+  - When the image for a step is specified via an input params, we should treat changes to the default value of that param in the same way as changes to a step image
+
+  In short, if task used in the pipeline is updated to a newer version, then corresponding pipeline needs to be bumped
+
+- The command+arg or script is changed (and the behavior) changes
+
+#### Questions
+
+- What should be the state of pipeline if a task used in the pipeline is deprecated ?
+
+### Ownership
+
+The ownership is controlled through the use of OWNERS file that allows
+the user listed in the file to approve PR to the sub-tree. e.g. OWNERS
+in the root directory controls the entire catalog while OWNERS file
+placed at /{resource-type}/{resource-name}/OWNERS is allowed to
+approve changes to all versions of the {resource-name}.
+
+- The root OWNERS file maps the tektoncd/catalog-maintainers
+  team. This team is responsible for the overall catalog, from
+  maintaining Resources (especially those who do not have a specific
+  set of OWNERS), making sure the testing infrastructure is healthy,
+  triaging issues, …
+
+- The *per-resource*OWNERS are responsible for maintaining their
+  Resource (updating images, behaviour, …)
+
+### Organization
+
+```bash=
+# Optional: owner(s) and reviewer(s) of the resource (all versions)
+./{resource-type}/{resource-name}/OWNERS
+# The README of the resource (what is the resource about, list of versions, …)
+./{resource-type}/{resource-name}/README.md
+
+# The resource itself
+./{resource-type}/{resource-name}/{version}/{resource}.yaml
+
+# The README of the versioned resource (usage, …)
+./{resource-type}/{resource-name}/{version}/README.md
+
+# The install script which will install all the related resources
+./{resource-type}/{resource-name}/{version}/install-script.sh
```

Comment:
why would each pipeline would need its own install script - installing a pipeline should be the same from pipeline to pipeline?

Reply:
This would help the user to just run the script and have the pipeline running where the script would include all the installation steps/commands for tasks and pipelines, so the user won't had to manually install a single resource everytime